wfsim/core/s2.py

```diff
@@ -383,13 +400,8 @@ def s2_pattern_map_diffuse(n_electron, z, xy, config, resource):
         :param resource: instance of the resource class
         """
         assert all(z < 0), 'All S2 in liquid should have z < 0'
-        
-        if config['enable_field_dependencies']['drift_speed_map']:
-            drift_velocity_liquid = resource.field_dependencies_map(z, xy, map_name='drift_speed_map')  # mm/µs
-            drift_velocity_liquid *= 1e-4  # cm/ns
-        else:
-            drift_velocity_liquid = config['drift_velocity_liquid']
-            
+        drift_velocity_liquid=S2.get_avg_drift_velocity(z, xy, config, resource)
```

Comment:
This should use the real xy, but the rest of timing estimation should use the observed one

Reply:
Is it like that? when I plot maps it looks like the intention is actually (x,y,z) true that gives average velocity.

Reply:
For the drift speed map you need the true (interaction) position, so this function should take it as it already does in the code. Then when you are sampling from maps such as the GARFIELD timing, sagging or S2 pattern map you should be using the field distorted position. Actually, you should be using the position of electrons at the liquid interface which is the field distorted one + the gate focusing effect, but I think this is within the posrec resolution and it is not a major probelm